reference/promotions.v3.yml

```diff
@@ -198,6 +198,10 @@ paths:
       operationId: getPromotionCodes
       parameters:
         - $ref: '#/components/parameters/PromotionIdPath'
+        - $ref: '#/components/parameters/BeforeCursorQuery'
+        - $ref: '#/components/parameters/AfterCursorQuery'
+        - $ref: '#/components/parameters/DeprecatedPageQuery'
```

Comment:
@bigcommerce/dev-docs do we need to do anything extra to call out the deprecation here ?

Reply:
@bc-vincent-zhao I think these descriptions look good and call out the deprecation.

<img width="645" alt="Screenshot 2024-10-29 at 8 52 58 AM" src="https://github.com/user-attachments/assets/34dc8ea0-948d-48d4-b8e1-da99674a6437">

Reply:
@bc-traciporter thanks, they look ok from schema perspective. I was more referring to if we need to send some proactive "deprecation notice" to users or include this as part of "change log" process ? so developers/merchants get notified about this changes in a timely manner. 🙇

Reply:
Oh, yes @bc-vincent-zhao we can do that. Can you send me a draft of the deprecation notice. I send out release notes every Thursday. Would you like to add something for tomorrow? When do you plan on releasing this change?

Reply:
>  I send out release notes every Thursday. 

thanks @bc-traciporter, this is good to know 😄 

we're still finalising some details on this, once we roll out the implementation, we will draft the deprecation notice and share it with you. cc @BC-SEven

Reply:
@bc-jwk, any thoughts? Do we want to deprecate offset pagination or continue to support both?

Reply:
@BC-SEven the best way forward would be to have an extended sunset period to minimise any radical changes to merchants, if we believe that the future is cursor-based pagination (from a technical perspective).

Can we organise it in such a way that:
1. We make updates to the dev documentation as usual with the newest changes
2. Include a deprecation note BUT with a deadline, e.g. 30 June 2025 or something realistic. That way it gives merchants and partners a heads-up that they will need to migrate their integration or risk having it "break".
3. We can then make a definitive call a few weeks or a month before the deadline. Maybe we can then identify if anyone is still using the whole method? Either way, I don't think it's too big of a problem since we've given plenty of notice.

@bc-traciporter I'm happy to create the deprecation notice. Do you have any previous examples I can model it off?

Reply:
Hi @bc-jwk Here are some examples of deprecation documentation. 
https://developer.bigcommerce.com/docs/storefront/stencil/cli/install
https://bcdevchangelog.noticeable.news/publications/api-updates-for-september-19-2024
I remember with this issue, the notice scared a lot of merchants and we received a lot of feedback. There were a lot of questions and people were nervous. I hope we can give clear instructions so as not to alarm merchants.

Reply:
Thanks @bc-traciporter 

@BC-SEven @bc-vincent-zhao can I please get 2 or 3 dot point reasons as to the benefits of switching to cursor-based pagination? We want to inform users and partners of the value of the switch so they understand that it's in their best interests to changeover.

Reply:
@bc-jwk the main benefit is to allow efficient and performant way iterate through a very large dataset

another somewhat controversial one is it allows users to build simper UI, without need to care about the pages / count.

Reply:
Thanks @bc-vincent-zhao.

@bc-traciporter please find a sample of what the notification can look like:
![image](https://github.com/user-attachments/assets/ab229ed1-a539-4a7b-a2c1-2afbd4df16d4)

The text is
`Introducing cursor-based pagination

This endpoint now supports cursor-based pagination for improved query performance. As each store can support millions of coupon codes, it is a more efficient way to iterate through these very large datasets, greatly reducing response times.

While existing implementations will continue to function as normal, it is recommended that integrations switch over to cursor-based pagination for maximum benefit.

We will begin to sunset offset-based pagination from 30 June 2025 to reduce maintenance costs and focus our efforts in fine-tuning the new standard. Please contact [support@bigcommerce.com](mailto:support@bigcommerce.com) for any questions`

Reply:
@bc-jwk Here is some feedback from @robincoferbc and the support team.

I'm not sure there would be any specific instructions for the cursor based pagination, but it would probably help to have a short and concise summary of the "breaking change" and what it means for merchants. If Promo team wants to provide official positioning/messaging, I'd love to include it in our [doc](https://bigcommercecloud.atlassian.net/wiki/spaces/clientsuccess/pages/1334378525/PSE+Handling+contact+support+or+escalation+references+in+public+docs).

Reply:
Hi @bc-traciporter thanks for the heads-up. Can you point me to where the feedback is? I can respond to it directly. Or should I just drop a note here? Not sure how often the support team check out GitHub.

Reply:
@bc-jwk Go to https://bigcommercecloud.atlassian.net/wiki/spaces/clientsuccess/pages/1334378525/PSE+Handling+contact+support+or+escalation+references+in+public+docs

Reply:
Hi @bc-traciporter had a Slack chat with @robincoferbc and we've squared it off. He's happy for the notice to go ahead from a Support perspective. Last thing to confirm is the sunset date.

@bc-vincent-zhao @BC-SEven is **30 June 2025** sufficient enough time for deprecation?

Reply:
yep, sounds good 👍